figma-shop/src/app/pages/shopping-cart/shopping-cart/shopping-cart.component.ts

```diff
@@ -13,34 +15,34 @@ export class ShoppingCartComponent implements OnInit {
   public personForm: FormGroup;
   public countries: string[] = countryList;
   public filteredCountries: Observable<string[]>;
-  public countriesFormControl = new FormControl();
-  constructor() { }
+  public countriesFormControl = new FormControl('', [Validators.required]);
+  public totalPrice: number;
+  public countStars = [1, 2, 3, 4, 5];
+  public orderedProducts: ProductInterface[];
+
+  constructor(
+    private orderService: OrderService) { }
 
   ngOnInit(): void {
     this.personForm = this.initPersonForm();
     this.filteredCountries = this.getFilteredCountries();
-    this.onFormChange();
+    this.orderedProducts = this.getProductFromLocal();
+    this.getTotal();
   }
 
   private initPersonForm(): FormGroup {
     return new FormGroup({
-      firstName: new FormControl(),
-      lastName: new FormControl(),
-      email: new FormControl(),
-      phone: new FormControl(),
-      address: new FormControl(),
+      firstName: new FormControl('', [Validators.required, Validators.minLength(2)]),
+      lastName: new FormControl('', [Validators.required, Validators.minLength(2)]),
+      email: new FormControl('', [Validators.required, Validators.pattern('^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$')]),
+      phone: new FormControl('', [Validators.required, Validators.pattern('((\\+38)?\\(?\\d{3}\\)?[\\s\\.-]?(\\d{7}|\\d{3}[\\s\\.-]\\d{2}[\\s\\.-]\\d{2}|\\d{3}-\\d{4}))')]),
```

Comment:
Please, add regex to variable and find the sematic name

Reply:
create 2 vriables, emailRegExp and phoneRegExp